npe2/_from_npe1.py

```diff
@@ -98,102 +106,112 @@ def plugin_packages() -> List[PluginPackage]:
-    for attr in ("package_name", "ep_name", "top_module"):
-        for p in plugin_packages():
-            if name == getattr(p, attr):
-                return p.package_name
-    raise KeyError(  # pragma: no cover
-        f"Unable to find a locally installed package for plugin {name!r}"
-    )
-
-
-@lru_cache()
-def npe1_plugin_manager() -> Tuple[PluginManager, Tuple[int, list]]:
-    pm = PluginManager("napari", discover_entry_point=NPE1_EP)
-    pm.add_hookspecs(HookSpecs)
-    result = pm.discover()
-    return pm, result
-
-
-def norm_plugin_name(plugin_name: Optional[str] = None, module: Any = None) -> str:
-    """Try all the things we know to detect something called `plugin_name`."""
-    plugin_manager, (_, errors) = npe1_plugin_manager()
-
-    # directly providing a module is mostly for testing.
-    if module is not None:
-        if plugin_name:  # pragma: no cover
-            warnings.warn("module provided, plugin_name ignored")
-        plugin_name = getattr(module, "__name__", "dynamic_plugin")
-        if not plugin_manager.is_registered(plugin_name):
-            plugin_manager.register(module, plugin_name)
-        return cast(str, plugin_name)
-
-    if plugin_name in plugin_manager.plugins:
-        return cast(str, plugin_name)
-
-    for pkg in plugin_packages():
-        for a, b in pkg.name_pairs:
-            if plugin_name == a and b in plugin_manager.plugins:
-                return b
-
-    # we couldn't find it:
-    for e in errors:  # pragma: no cover
-        if module and e.plugin == module:
-            raise type(e)(e.format())
-        for pkg in plugin_packages():
-            if plugin_name in (pkg.ep_name, pkg.package_name, pkg.top_module):
-                raise type(e)(e.format())
-
-    msg = f"We tried hard! but could not detect a plugin named {plugin_name!r}."
-    if plugin_manager.plugins:
-        msg += f" Plugins found include: {list(plugin_manager.plugins)}"
-    raise metadata.PackageNotFoundError(msg)
-
-
 def manifest_from_npe1(
-    plugin_name: Optional[str] = None, module: Any = None
+    plugin: Union[str, metadata.Distribution, None] = None,
+    module: Any = None,
+    shim=False,
 ) -> PluginManifest:
-    """Return manifest object given npe1 plugin_name or package name.
+    """Return manifest object given npe1 plugin or package name.
 
-    One of `plugin_name` or `module` must be provide.
+    One of `plugin` or `module` must be provide.
 
     Parameters
     ----------
-    plugin_name : str
-        Name of package/plugin to convert, by default None
-    module : Module
+    plugin : Union[str, metadata.Distribution, None]
+        Name of package/plugin to convert.  Or a `metadata.Distribution` object.
+        If a string, this function should be prepared to accept both the name of the
+        package, and the name of an npe1 `napari.plugin` entry_point. by default None
+    module : Optional[Module]
```

Comment:
Type in docstring does not match type in signature, maybe we can go "ModuleType" 

We can always change later to not mess up the 3 PR chains.